### DISCUSSION TITLE
Keywords to disable notifications

### DISCUSSION THREAD

**Discussion by raphj** [element-hq/element-meta#690] [Ideas]
#### What would you like to do?

I'd like to be able to define keywords or regular expressions that will disable the message notification (possibly for some channel specifically). And maybe a maximum message length (if the message is longer, produce the notification). It would be like the  notification keywords that already exist, but opposite.

#### Why would you like to do it?

I have notification enabled for each message. This is usually fine because the channels I'm in are quite low traffic and I want to be able to passively follow discussions while doing something else. However, there are messages that don't warrant any notifications. There's no way I'd meaningfully react to such messages and those can wait the next time I open Element.

For instance, it is the birthday of somebody and people are used to say "Happy birthday X!" in a specific channel. This is nice and warm, it also does not require being notified. The birthday lasts for the day, the messages can wait a few minutes or hours to be read.

#### How would you like to achieve it?

I'd like to define a list of keywords or sub-strings like "Happy birthday" that will disable the notification for messages containing it, possibly restricted to a channel. It's not perfect, because I could miss critical information in message like this: "Happy birthday X! By the way, [CRITICAL INFORMATION]" so a regular expression or defining a maximum message length could be useful but then again, the critical information will be seen later and maybe it should belong to a separate message.  

### Have you considered any alternatives?

I considered temporarily disabling notifications for the channel in which these messages are sent but I still want to be notified for the other messages in this channel.

**Comment by t3chguy** [comment]
Notifications are handled by both clients & servers, I suggest opening this issue on https://github.com/matrix-org/matrix-spec

**Reply by Mikaela** [reply]
Seems to be https://github.com/matrix-org/matrix-spec/issues/1074 in case anyone is trying to find it from here

**Comment by kittykat** [comment]
Hi, I'm going to move this issue to Element meta discussions as we need to consider it cross-platform :+1:

**Comment by wotography** [comment]
just found this now, but already opened a feature request here for this: [https://github.com/element-hq/element-x-ios/issues/6031](https://github.com/element-hq/element-x-ios/issues/6031)
with the addition, that it keeps the chat in its original position, not moving it up in the "latest chat list".
